src/components/TextInput.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { Field } from 'redux-form';
+import TextField from 'material-ui/TextField';
+import PropTypes from 'prop-types';
+
+function RenderTextField({ ...props }) {
+  const textProp = Object.assign({}, props);
+  delete textProp.meta;
+  delete textProp.input;
+
+  return (
+    <TextField
+      {...textProp}
+    />
+  );
+}
+const TextInput = ({ name, ...props }) => (
+  <Field
+    {...props}
+    name={name}
+    id={name}
```

Comment:
hmm, I don't see an `id` being used in the material-ui examples, and also not in the redux-form examples. It should not be required, I wonder where those warnings come from.

`id` should not really be used in React components like this—it's just too hard to ensure that a given name is unique.